int32 allied_type
---
int32 intensity_auto_precedence
int32 error